relevanceai/vector_tools/plot_txt_gist.py

```diff
@@ -0,0 +1,202 @@
+                docs.extend(batch_doc['documents'])
+        else:
+            docs = self.client.get_all_documents(dataset_id = self.dataset_id, filters = filters, select_fields = fields)
+        return docs
+
+    def batch_load_docs(self, fields:List[str], filters:List[dict] = [], page_size:int = 200, cursor:str = None):
+        return self.client.datasets.documents.get_where(dataset_id=self.dataset_id, select_fields=fields, 
+                page_size= page_size, filters=filters, cursor = cursor)
+
+    def kmeans_clustering(self, docs, vector_field = str, k : int = 10, alias: str = "kmeans"):
+        print(" * Kmeans Clustering")
+        clusterer = KMeans(k = k)
+        clustered_docs = clusterer.fit_documents([vector_field], docs, alias = alias)
+        res = self.client.update_documents(self.dataset_id, clustered_docs, chunksize = self.upload_chunksize)
+        centers = clusterer.get_centroid_docs()
+        self.client.services.cluster.centroids.insert(dataset_id = self.dataset_id, cluster_centers = centers,
+                vector_field = vector_field, alias = alias)
+        return centers, clustered_docs
+
+    @staticmethod
+    def get_field_value_by_id(docs:List[dict], id:str, field):
+        for doc in docs:
+            if doc['_id'] == id:
+                return doc[field]
+
+    def get_cluster_datafield(self, docs:List[dict], vector_field: str, text_field:str, alias: str,
+        lower:bool = True, remove_digit:bool = True, remove_punct: bool = True):
+        print(" * Extracting cluster info")
+        cluster_data = {}
+        for i, doc in tqdm(enumerate(docs)):
+            cluster_name = doc['_'.join([self.cluster_field, vector_field, alias])]
+            if cluster_name not in cluster_data:
+                cluster_data[cluster_name] = {'data':[]}
+            text = self.normalize_text(
+                txt = self.get_field_value_by_id(docs, id = doc['_id'], field = text_field), 
+                lower = lower, remove_digit = remove_digit, remove_punct = remove_punct)
+            cluster_data[cluster_name]['data'].append(text)
+        return cluster_data
+
+    @staticmethod
+    def get_cluster_population(cluster_data:dict):
+        return [(cluster_name, len(v['data'])) for cluster_name, v in cluster_data.items()]
+
+    def get_cluster_word_freq(self, cluster_data: dict, remove_stop_words: bool = True, 
+        additional_stop_words: List[str] = [], cluster_representative_cnt : int= 3):
+        for cluster in tqdm(cluster_data):
+            cluster_data[cluster]['word_freq'] = self.get_word_frequency(str_list = cluster_data[cluster]['data'], 
+                remove_stop_words = remove_stop_words, additional_stop_words = additional_stop_words)
+        return {c:[cluster_data[c]['word_freq'][:cluster_representative_cnt]] for c in cluster_data}  
+
+    def dim_reduction(self, vector_data: List, embedding_dims=2, k=800, n_epochs_without_progress=100, fit = True):
```

Comment:
`fit` is a strange argument to be called "True", Maybe "run_fit"? Or maybe something starting with `has` or `is`